packages/edit-post/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -7,37 +7,64 @@ import { get } from 'lodash';
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
-import { Button } from '@wordpress/components';
+import { Button, Icon } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 import { wordpress } from '@wordpress/icons';
 
 function FullscreenModeClose() {
-	const { isActive, postType } = useSelect( ( select ) => {
-		const { getCurrentPostType } = select( 'core/editor' );
-		const { isFeatureActive } = select( 'core/edit-post' );
-		const { getPostType } = select( 'core' );
+	const { isActive, isRequestingSiteIcon, postType, siteIconUrl } = useSelect(
+		( select ) => {
+			const { getCurrentPostType } = select( 'core/editor' );
+			const { isFeatureActive } = select( 'core/edit-post' );
+			const { isResolving } = select( 'core/data' );
+			const { getEntityRecord, getPostType } = select( 'core' );
+			const siteData =
+				getEntityRecord( 'root', '__unstableBase', undefined ) || {};
 
-		return {
-			isActive: isFeatureActive( 'fullscreenMode' ),
-			postType: getPostType( getCurrentPostType() ),
-		};
-	}, [] );
+			return {
+				isActive: isFeatureActive( 'fullscreenMode' ),
+				isRequestingSiteIcon: isResolving( 'core', 'getEntityRecord', [
+					'root',
+					'__unstableBase',
+					undefined,
+				] ),
+				postType: getPostType( getCurrentPostType() ),
+				siteIconUrl: siteData.site_icon_url,
+			};
+		},
+		[]
+	);
 
 	if ( ! isActive || ! postType ) {
 		return null;
 	}
 
+	let buttonIcon = <Icon size="36px" icon={ wordpress } />;
+
+	if ( siteIconUrl ) {
+		buttonIcon = (
+			<img
+				alt={ __( 'Site Icon' ) }
+				className="edit-post-fullscreen-mode-close_site-icon"
+				src={ siteIconUrl }
+			/>
+		);
+	} else if ( isRequestingSiteIcon ) {
+		buttonIcon = null;
```

Comment:
I'm wondering if it would make sense to display the WP logo while the site icon is loading, instead of showing nothing. 🤔 

Something like:
```jsx
const buttonIcon = siteIconUrl
	? <img src={ siteIconUrl } />
	: <Icon icon={ wordpress } />
```

I guess we wouldn't even need `isRequestingSiteIcon`, as `siteIconUrl` should be falsy while loading.

Reply:
I'm on the fence about this one. A similar thought had occurred to me, and it would certainly simplify the code, but then again: 

- UX: The WP logo might actually confuse people ("Hey, I set my own logo, why is it still showing the W thing?")
- IIUC, we'd be showing the placeholder while `siteIconUrl` is loading (rather than while the actual image file is being loaded from that URL).
- We're actually preloading the endpoint that gives us `siteIconUrl`, so the URL should be available near-instantly.

My vote for keeping as-is; this is something we can iterate on later.

Reply:
Makes sense! I'll resolve the conversation then 🙂